### PR TITLE
scripts: add JS-only script for large cores on Mac

### PR DESCRIPTION
**Pull request by indutny**
Fix: #68

### CODE REVIEW

**Comment by indutny** [comment]
This is a very limited script as it supports only 64-bit Little Endian platforms. We can potentially expand it later if Apple won't update lldb earlier :wink:

File: scripts/macho2segments.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env node
+'use strict';
+
+const assert = require('assert');
+const fs = require('fs');
+const Buffer = require('buffer').Buffer;
```
Comment (cjihrig) [inline scripts/macho2segments.js]: We don't really need to require this, right?

File: scripts/macho2segments.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  return;
+}
+
+const fd = fs.openSync(process.argv[2], 'r');
+
+function read(start, end) {
+  const data = Buffer.alloc(end - start);
+  const bytesRead = fs.readSync(fd, data, 0, data.length, start);
+
+  assert.equal(data.length, bytesRead, 'Read less than expected');
```
Comment (cjihrig) [inline scripts/macho2segments.js]: Any reason not to use `strictEqual()` throughout this file?

File: scripts/macho2segments.js (hunk 1 above)
Comment (indutny) [inline scripts/macho2segments.js]: That's how I always roll :) Maybe one day we will be able to remove global `Buffer`.

File: scripts/macho2segments.js (hunk 2 above)
Comment (indutny) [inline scripts/macho2segments.js]: Ack.

**Comment by indutny** [comment]
@cjihrig your feedback is very welcome!

File: scripts/macho2segments.js (hunk 1 above)
Comment (cjihrig) [inline scripts/macho2segments.js]: Very wishful thinking :-)

**Comment by hhellyer** [comment]
@indutny - I compared it to the output from the otool script on a smaller dump. It gets the section sizes wrong due to some the string concatenation in parseUInt64LE.

I switched to
```
function parseUInt64LE(buf, off) {
  var low = buf.readUInt32LE(off).toString(16);
  var high = buf.readUInt32LE(off + 4).toString(16);
  while (high.length < 8)
    high = '0' + high;
  while (low.length < 8)
    low = '0' + low;
  return high+low;
}
```
and got matching output.

That needs fixing but I'm pretty impressed with how quickly you parsed it otherwise!

It also picks up a couple of extra ranges that otool doesn't and misses off the leading 0x from the hex output. The ranges are probably fine - otool probably has a reason to hide them but it probably doesn't affect us and if C++ std::hex doesn't mind missing the leading 0x neither do I.

**Comment by indutny** [comment]
@hhellyer fixed mentioned nits, PTAL.

**Comment by indutny** [comment]
Landed in 1a27c72, thank you!

**Pull request closed without merging**